jib-core/src/main/java/com/google/cloud/tools/jib/docker/ImageTarball.java

```diff
@@ -61,6 +68,55 @@ public ImageTarball(
   }
 
   public void writeTo(OutputStream out) throws IOException {
+    if (image.getImageFormat() == OCIManifestTemplate.class) {
+      ociWriteTo(out);
+    } else {
+      dockerWriteTo(out);
+    }
+  }
+
+  private void ociWriteTo(OutputStream out) throws IOException {
+    TarStreamBuilder tarStreamBuilder = new TarStreamBuilder();
+    OCIManifestTemplate manifest = new OCIManifestTemplate();
+
+    // Adds all the layers to the tarball and manifest
+    for (Layer layer : image.getLayers()) {
+      DescriptorDigest digest = layer.getBlobDescriptor().getDigest();
+      long size = layer.getBlobDescriptor().getSize();
+
+      tarStreamBuilder.addBlobEntry(layer.getBlob(), size, "blobs/sha256/" + digest.getHash());
+      manifest.addLayer(size, digest);
+    }
+
+    // Adds the container configuration to the tarball and manifest
+    JsonTemplate containerConfiguration =
+        new ImageToJsonTranslator(image).getContainerConfiguration();
+    BlobDescriptor configDescriptor =
+        Blobs.from(containerConfiguration).writeTo(ByteStreams.nullOutputStream());
+    manifest.setContainerConfiguration(configDescriptor.getSize(), configDescriptor.getDigest());
+    tarStreamBuilder.addByteEntry(
+        JsonTemplateMapper.toByteArray(containerConfiguration),
+        "blobs/sha256/" + configDescriptor.getDigest().getHash());
+
+    // Adds the manifest to the tarball
+    BlobDescriptor manifestDescriptor =
+        Blobs.from(manifest).writeTo(ByteStreams.nullOutputStream());
+    tarStreamBuilder.addByteEntry(
+        JsonTemplateMapper.toByteArray(manifest),
+        "blobs/sha256/" + manifestDescriptor.getDigest().getHash());
+
+    // Adds the oci-layout and index.json
+    tarStreamBuilder.addByteEntry(
+        "{\"imageLayoutVersion\": \"1.0.0\"}".getBytes(StandardCharsets.UTF_8), "oci-layout");
```

Comment:
This looks a little hacky, but I'd rather just write this string directly than implement another `JsonTemplate`.